templates/terraform/constants/pubsub_topic.go.erb

```diff
@@ -0,0 +1,17 @@
+<%- # the license inside this block applies to this file
+	# Copyright 2020 Google Inc.
+	# Licensed under the Apache License, Version 2.0 (the "License");
+	# you may not use this file except in compliance with the License.
+	# You may obtain a copy of the License at
+	#
+	#     http://www.apache.org/licenses/LICENSE-2.0
+	#
+	# Unless required by applicable law or agreed to in writing, software
+	# distributed under the License is distributed on an "AS IS" BASIS,
+	# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+	# See the License for the specific language governing permissions and
+	# limitations under the License.
+-%>
+func pubsubPollWaiter(res map[string]interface{}) PollWaiter {
+	return &ExistancePollWaiter{}
```

Comment:
This method appears to be a way to alias from `{client_name_camel}PollWaiter` to a generic poll waiter. Would it make more sense to just move this into the asyncPoll configuration?

Reply:
that's what I originally did, but I don't know whether it makes sense to add it to async.rb which technically applies to all products but really only works for terraform? and then in terraform we have to explicitly add it as a new override property async (or something like a poll waiter property, which is worse) to fix it

Reply:
I agree that adding async overrides is probably overkill. Also TBH most of the existing async configuration is only used by ansible but it's all conceptually something that could be provider agnostic (eg: full_url, wait_ms, etc). But feel free to override.

Reply:
i decided this morning i'm going to try to override it - i'll ping you when im ready for review again?